src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,99 @@
-import { useState } from 'react';
+import { useMemo, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [description, setDescription] = useState('');
+  const [title, setTitle] = useState('');
+  const [imgUrl, setImgUrle] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const isError = useMemo(() => (
+    !title.trim()
+    || !imgUrl.trim()
+    || !imdbId.trim()
+    || !imdbUrl.trim()
+  ), [
+    title,
+    imdbId,
+    imdbUrl,
+    imgUrl,
+  ]);
+
+  const resetForm = () => {
+    setTitle('');
+    setDescription('');
+    setImgUrle('');
+    setImdbUrl('');
+    setImdbId('');
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setCount(prevCount => prevCount + 1);
+
+    onAdd({
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    });
+
+    resetForm();
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={handleSubmit}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={setTitle}
         required
       />
 
       <TextField
         name="description"
         label="Description"
-        value=""
+        value={description}
+        onChange={(value) => setDescription(value)}
       />
 
       <TextField
         name="imgUrl"
         label="Image URL"
-        value=""
+        value={imgUrl}
+        onChange={(value) => setImgUrle(value)}
```

Comment:
Still actual - try to rewrite in all TextField this way 👇
In line 66 you have this example.
```suggestion
        onChange={setImgUrle}
```